bundle/edu.gemini.pit/src/main/scala/edu/gemini/pit/ui/view/obs/ObsListView.scala

```diff
@@ -666,54 +669,78 @@ class ObsListView(shellAdvisor:ShellAdvisor, band:Band, queueLookup: Target => U
 
     val lens = olmLens
 
+    // General fix-it method.
+    def fixer(findAndProcess: ObsListElem => Boolean): Unit = {
+      model.foreach(_.elems.find(findAndProcess))
+    }
+
     // Public method for fix-its
-    def fixEmpty[A](grouping:ObsListGrouping[A]) {
-      model.foreach { m =>
-        m.elems.find {
-          case g:ObsGroup[_] if (g.grouping == grouping) && g.grouping.get(g).isEmpty =>
-            viewer.selection = Some(g)
-            true
-          case _                                                                      => false
-        }
-      }
+    def fixEmpty[A](grouping:ObsListGrouping[A]) = fixer {
+      case g: ObsGroup[_] if (g.grouping == grouping) && g.grouping.get(g).isEmpty =>
+        viewer.selection = Some(g)
+        true
+      case _ => false
+    }
+
+    // Add missing conditions.
+    def addConditions() = fixer {
```

Comment:
Now we have `addConditions`, `addBlueprint`, and `addTarget` to add missing elements to an observation.

This doesn't work with obs time, as that dialog is treated differently and can only be filled in after other parameters are, so that is still just highlighted if the message that it is missing is double clicked.